esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -193,13 +192,14 @@ impl<'a> PartitionTable<'a> {
         }
 
         let mut raw = Self {
-            binary,
+            // we checked binary before
+            binary: unsafe { core::mem::transmute::<&[u8], &[[u8; 32]]>(binary) },
```

Comment:
Wait, how does this work? What will be the slice's length after the transmute? Is the compiler smart enough to divide by 32? Also, that 32 should be `RAW_ENTRY_LEN`.

Reply:
uh you are right - it's not smart enough 👍

Reply:
Ah crap, https://doc.rust-lang.org/beta/core/primitive.slice.html#method.as_chunks  is still unstable

Reply:
Yeah transmute is a sharp tool - we can go through slice::from_raw_parts for now

Reply:
https://doc.rust-lang.org/beta/core/primitive.slice.html#method.align_to is also an option